File: DinnersAccounting/UI/Scripts/Model/OrderDetail.js
//------------------------ORDER_DETAILS---------------------------------------------------
var OrderDetail = function (id, product, quantity) {
    var self = this;
    self.Id = id;
    self.Product = product;
    self.Quantity = ko.observable(quantity);

    return {
        Id: self.Id,
        Product: self.Product,
        Quantity: self.Quantity
    };
};
//-----------------------------------------------------------------------------------------

File: DinnersAccounting/Domain/Entities/OrderDetail.cs
using DA.Dinners.Model;

namespace DA.Dinners.Domain
{
    /// <summary>
    /// Describe single item of an order : product itme and quantity
    /// </summary>
    public class OrderDetail
    {
        public int Id { get; set; }

        /// <summary>
        /// Gets or sets the quantity of product in order
        /// </summary>
        /// <value>
        /// The quantity.
        /// </value>
        public int Quantity { get; set; }

        /// <summary>
        /// Gets or sets the product instance
        /// </summary>
        /// <value>
        /// The product.
        /// </value>
        public virtual Product Product { get; set; }

        public virtual Order Order { get; set; }
    }
}